lib/views/pages/home/content/ranking/ranking_card.dart

```diff
@@ -0,0 +1,128 @@
+import "package:flutter/material.dart";
+import "package:proxima/models/ui/ranking/ranking_element_details.dart";
+import "package:proxima/models/ui/user_avatar_details.dart";
+import "package:proxima/views/components/content/publication_header/user_profile_pop_up.dart";
+import "package:proxima/views/components/content/user_avatar/user_avatar.dart";
+
+/// A widget that displays a ranking card.
+/// The card displays the user's rank, display name, and centauri points.
+/// The card is clickable and opens a user profile pop-up.
+/// The card's color can be customized.
+class RankingCard extends StatelessWidget {
+  static const userRankrankTextKey = Key("userRankrankText");
+  static const userRankDisplayNameTextKey = Key("userRankDisplayNameText");
+  static const userRankCentauriPointsTextKey =
+      Key("userRankCentauriPointsText");
+  static const userRankAvatarKey = Key("userRankAvatar");
+
+  // Sized box for the rank text to make sure that the text is centered
+  // and that surrounding element spacing is consistent.
+  // This width is allow to display three digits.
+  static const _sizedBoxRankWidth = 35.0;
+
+  static const _rankOneColor = Color.fromARGB(255, 255, 218, 75);
+  static const _rankSecondColor = Color.fromARGB(255, 217, 218, 204);
+  static const _rankThirdColor = Color.fromARGB(255, 229, 153, 137);
+
+  // This text is used when the parameter [userRank] of the
+  // provided [RankingElementDetails] is null.
+  static const _nullRankText = "---";
+
+  const RankingCard({
+    super.key,
+    required this.rankingElementDetails,
+  });
+
+  final RankingElementDetails rankingElementDetails;
+
+  @override
+  Widget build(BuildContext context) {
+    /// On tap function to open the user profile pop-up.
+    onTapUserPopUp() => {
+          showDialog(
+            context: context,
+            builder: (BuildContext context) => UserProfilePopUp(
+              displayName: rankingElementDetails.userDisplayName,
+              username: rankingElementDetails.userUserName,
+              centauriPoints: rankingElementDetails.centauriPoints,
+            ),
+          ),
+        };
+
+    final cardColor = switch (rankingElementDetails.userRank) {
```

Comment:
Here I think it would make more sense if we had colors in a `static` `const` list (`[]`) in the order of the rank and then we can query the right color directly using `rankingColors.elementAtOrNull(rankingElementDetails.userRank - 1)`.

What do you think ?

Reply:
I think this is a good idea, but since the `userRank` can be null it really complexify the code as we have
```dart
static const _rankingColors = [
    (
      Color.fromARGB(255, 255, 218, 75),
      Color.fromARGB(255, 206, 176, 60)
    ), // Rank 1
    (
      Color.fromARGB(255, 217, 218, 204),
      Color.fromARGB(255, 161, 161, 151)
    ), // Rank 2
    (
      Color.fromARGB(255, 229, 153, 137),
      Color.fromARGB(255, 138, 89, 80)
    ), // Rank 3
  ];

```

But doing that:

```dart 
final cardColorTuple = _rankingColors
        .elementAtOrNull(rankingElementDetails.userRank - 1);
final cardColor = isLightMode ? cardColorTuple?.$1 : cardColorTuple?.$2;
```
does not work as userRank can be null.

So we need to either handle the null case before getting the cardColor. Or doing something like:
```dart 
final cardColorTuple = _rankingColors.elementAtOrNull(
      (rankingElementDetails.userRank ?? _rankingColors.length + 1) - 1,
    );

```

Which I think is not ideal and really degrade the readability of the code.

What do you think ?

Reply:
What about simply:

```dart
final cardColorTuple = userRank != null ? _rankingColors.elementAtOrNull(userRank - 1) : null;
final cardColor = isLightMode ? cardColorTuple?.$1 : cardColorTuple?.$2;
```

Reply:
Yeah we can handle the null case with the `?`.

Added in: [d6d832a](https://github.com/ProximaEPFL/proxima/pull/322/commits/d6d832ab7a7422a0dca3d72dbc5aab1bb370ed15)

Thank you !